arm/ui/json_api.py

```diff
@@ -416,10 +416,14 @@ def abandon_job(job_id):
         return json_return
 
     try:
+        # Get job id
         job = Job.query.get(int(job_id))
+        # Stop job process id (pid)
         job.status = JobState.FAILURE.value
         job_process = psutil.Process(job.pid)
         job_process.terminate()  # or p.kill()
+        # release job from drive
+        job.drive.release_current_job()
```

Comment:
further below is the job.eject() which should also release the job. Would be good to fix job.eject() if it got issues instead of adding it here. I guess the main problem behind it was that [the current id did not get cleared](https://github.com/automatic-ripping-machine/automatic-ripping-machine/pull/1458/files#diff-61cb405a6a10f0ba81c1686aca959f76fe364dfbf780897a99a8b5f9a07d5c49R275).

Reply:
would it not be better to release the id post rejecting the job  before the eject?

Reply:
hm. I am not sure if I get you right. I think the naming of job.eject is a bit confusing here. It releases the job from the drive. So the drive is not associated to a job anymore. It also may trigger drive.eject.

I know that previously there were multiple locations where the drive / and or job got released/ejected and I just hope that we are not getting into confusing "eject" flow again.

Otherwise, it does not hurt to release the job here per se.

Reply:
Not quite right. Drive.eject releases the job in any case. The difference between job.eject and drive.eject is that drive.eject ejects the drive even if auto eject is not set. Job.eject is on job level and respects the autoeject var.